wauw

RobusB100 opened the pull request:
insane

bitphage committed:
Fix wrong rel in pytest.approx()

bitphage committed:
Simplify increase logic for least order

We don't need this complex logic now after implementing pre-calculations
of increases.

bitphage committed:
Accept small inaccuracy for full-sized closer order

This allows to fix tests failing due to #575. Note this is not a final
fix!

bitphage committed:
Fix SO tests

pytest.approx() calls were invoked with wrong "rel" thus causing too
much bias allowance. The error is now fixed along with ala tests
affected.

Closes: #651

xeroc commented:
Holy ... quite cool to see that much movement in dexbot.

Not sure it makes any sense to merge this upstream as stakemachine is currently unmaintained (no time)

bitphage committed:
Remove soft limit from place_closer_order()

This allows to place closer order on big side after an order was filled
on small side.

Closes: #588

joelvai committed:
Change dexbot version number to 0.15.3

bitphage committed:
Add initial table structure as alembic migration

This fixes fresh install when sqlite database is created from scratch.
Instaead of creating with `create_all()` we're dedicating this job to
alembic.

bitphage committed:
Fix test

bitphage committed:
Fix running migrations on windows

bitphage committed:
Package migrations as data files

Without this migrations files are not accessible in bundled versions.

joelvai committed:
Change dexbot version number to 0.15.4

bitphage committed:
Fix migrations tests

We're now creating table structure from migrations, this test needs to
be adjusted.

bitphage committed:
Add forgotten table field

bitphage committed:
Don't try to delete empty queryset

bitphage committed:
Revert "Fix migrations tests"

This reverts commit 1386d02e8ac5630db534c356ada9307c03eb5fb7.

bitphage committed:
Fix migrations applying for existing installations

bitphage committed:
Fix migrations access when packaged into egg

Use pkg_resources to access migrations files. When dexbot is packaged
into egg file (normal `setup.py install` result), we cannot just access
files inside egg (zip archive), so we're using native setuptools API to
extract the files (resource_filename()). In case files are not in egg,
pkg_resources will use normal file access.

https://setuptools.readthedocs.io/en/latest/pkg_resources.html#basic-resource-access
https://setuptools.readthedocs.io/en/latest/setuptools.html#accessing-data-files-at-runtime
http://peak.telecommunity.com/DevCenter/PythonEggs#accessing-package-resources

bitphage committed:
Add __init__.py to migrations

joelvai committed:
Change dexbot version number to 0.15.5

bitphage committed:
Fix required-participation value

bitphage committed:
Limit num_retries in tests

If we're failing to connect to the local node, give up quickly to avoid
build stuck.

bitphage committed:
Mark some tests as mandatory

This needed to run some tests from travis-ci.

bitphage committed:
Enable mandatory tests run in travis

bitphage committed:
Uninstall deve requirements after running tests

bitphage committed:
Remove unneded escaping

Escaping caused travis error

bitphage committed:
Call pip uniinstall with -y

g3d committed:
[#664] Add ping into image, as it's required for node checks

bitphage committed:
Set clear rules on dynamic CP conditions

bitphage committed:
Adjust expiration override in case of cp_from_last_trade

If custom expiration is set and it's too small, it makes a bad condition
for worker with cp_from_last_trade active. So just make sure we have
default expiration when cp_from_last_trade is used.

bitphage committed:
Detect SO -> RO transition and purge orders

Closes: #666

bitphage committed:
Preserve bootstrap state when CP from last trade

* if worker newly created, use market center price
* if worker is not new, always try to find own last trade even on startup

Closes: #672

bitphage committed:
Improve last trade search

Don't limit to 1 op, get all ops and iterate to find first match. This
is a preparation to run multiple workers on same account.

bitphage committed:
Fix return type in get_martket_xxx_price

bitphage committed:
Add more logging for center price

bitphage committed:
Fix typo

bitphage committed:
Add check for zero spread

KOTH worker must avoid filling of opposite orders in case spread is too
small. This commit adds a check for this.

Closes: #668

bitProfessor committed:
RO tests

bitphage committed:
Don't disable worker in case of 0 balance

When KOTH is used in "both" mode, but has only one asset, don't disable
worker, just wait until some balance will be available.

Closes: #670

bitphage committed:
Fix insufficient balance error

There was a flaw in logic which allowed to bypass a check for minimal
allowed order size. As a result, worker tried to place too small order
causing an error `bitsharesapi.exceptions.UnhandledRPCError: Assert
Exception: d.get_balance( *_seller, *_sell_asset ) >= op.amount_to_sell:
insufficient balance`

Closes: #649

PermieBTS committed:
Update README.md

PermieBTS committed:
Update README.md

bitphage committed:
Recalculate price for virtual orders

Order price is now more precise according to actual asset precisions.

Closes: #642

joelvai committed:
Merge remote-tracking branch 'vvk/travis-mandatory-tests' into pr-663-merge

joelvai committed:
Change dexbot version number to 0.15.6

joelvai committed:
Merge remote-tracking branch 'vvk/668-670-koth-fixes' into pr-683-merge

joelvai committed:
Change dexbot version number to 0.15.7

joelvai committed:
Merge remote-tracking branch 'g3d/hotfix/664-ping-in-docker-container' into pr-665-merge

joelvai committed:
Change dexbot version number to 0.15.8

joelvai committed:
Merge remote-tracking branch 'thehapax/596-properties' into pr-623-merge

joelvai committed:
Revert shared_bitshares_instance() back

This caused WalletLocked error when switching nodes. For now better
leave it there.

joelvai committed:
Change dexbot version number to 0.15.9

joelvai committed:
Merge remote-tracking branch 'vvk/651-SO-wrong-rel-in-approx' into pr-684-merge

joelvai committed:
Change dexbot version number to 0.15.10

bitphage committed:
Merge remote-tracking branch 'upstream/devel' into single-account-multiple-workers

bitphage committed:
Update docstring

bitphage committed:
Add SO test for single account multiple workers

bitProfessor committed:
fixed duplicate name test_correct_asset_names

bitphage committed:
Merge remote-tracking branch 'upstream/devel' into ro-tests

bitphage committed:
Merge branch 'ro-refactor' into ro-tests

bitphage committed:
Refactor RO tests

bitphage committed:
Avoid codacy warnings

We're not using random.xxx() for security purposes.

joelvai committed:
Change dexbot version number to 0.16.0

joelvai committed:
Merge remote-tracking branch 'vvk/single-account-multiple-workers' into pr-441-merge

joelvai committed:
Change dexbot version number to 0.17.0

joelvai committed:
Add python version print before Linux build

joelvai committed:
Update MacOS to use Python 3.7

joelvai committed:
Change dexbot version number to 0.17.1

bitphage committed:
Fix race condition in SO

A rare race condition could occur when increasing an order and replacing
same order as partially filled.

bitphage committed:
Fix getting last own trade price in RO

Add an explicit check if trade belongs to current market. This fixes a
situation when account is used by multiple workers with asset
intersections. Thus a wrong trade (from different market) may be used
resulting in incirrect price and further trades.

thehapax committed:
fix requirements so compatible with graphene and uptick

joelvai committed:
Change dexbot version number to 0.17.2

joelvai committed:
Merge remote-tracking branch 'vvk/fix-RO-last-trade-price' into pr-699-merge

joelvai committed:
Change dexbot version number to 0.17.3

joelvai committed:
Merge remote-tracking branch 'vvk/fix-SO-race-condition' into pr-697-merge

joelvai committed:
Change dexbot version number to 0.17.4

joelvai committed:
Merge remote-tracking branch 'vvk/642-multiple-orders-same-price' into pr-690-merge

joelvai committed:
Change dexbot version number to 0.17.5

thehapax committed:
add cancel all market orders option to cli. change host_ip from 8.8.8.8 to 1.1.1.1

thehapax committed:
fix pep8

thehapax committed:
fix codacy syntax error

bitphage committed:
Fix cancel dust order

A slight regression was introduced in 32dbafbcace9d1113578416cc24041d75eea3741

froooze committed:
make lower/upper bound more clear

bitphage committed:
Add log message

bitphage committed:
Fix using last own trade price as center price

Price calculation was wrong when last trade was a taker trade (this
happens naturally when 3rd party limit order overlaps with own new
order).

Closes: #708

thehapax committed:
add reset nodelist to default, sort if option given

thehapax committed:
amend usage message

thehapax committed:
move comments

thehapax committed:
fix method name reset_nodes to be different in cli

dls-cipher committed:
Update config.py

Removal/addition of bitshares.org nodes

Note: Not from worker anymore, so direct contact @murda_ra on Telegram for issues

joelvai committed:
Change dexbot version number to 0.17.6

joelvai committed:
Merge remote-tracking branch 'vvk/708-fix-get-own-last-trade' into pr-709-merge

joelvai committed:
Change dexbot version number to 0.17.7

joelvai committed:
Merge remote-tracking branch 'vvk/fix-SO-dust-order-cancel' into pr-705-merge

joelvai committed:
Change dexbot version number to 0.17.8

bitphage committed:
Add pre-commit hooks

Hooks ensure proper formatting and linting before commiting

bitphage committed:
Adjust docstring to fix rendering

bitphage committed:
Remove reference to non-existent docs

bitphage committed:
Remove statemachine mentiones from docs

bitphage committed:
Fix price inversion for virtual sell orders

bitphage committed:
Add workaround for test_maintain_strategy_one_sided

bitphage committed:
Add switch to disable fallback logic in SO

bitphage committed:
Close spread before replacing virtual orders with real

bitphage committed:
Add workaround to fix some tests

Problem described here https://github.com/Codaone/DEXBot/issues/575
Workaround is to use approx value for spread comparison accepting a
slight error.

bitphage committed:
Remove calculation of asset thresholds

Set asset threshold when increasing orders only. Threshold now is a min
amount required to increase an order. There is a balance check in
place_closer_xxx_order(), so
https://github.com/Codaone/DEXBot/issues/554 should not appear.

Closes: #554

bitphage committed:
Implement Stop Loss in SO strategy

bitphage committed:
Append orderid when reconstructing order data

bitphage committed:
Add check_last_run decorator

bitphage committed:
Fix proper order reconstruction

calculate_order_data() should reconstruct buy/sell orders differently to
achive same result as return from the node (without inversion)

bitphage committed:
Fix operational depth when fallback logic is off

bitphage committed:
Add cli command to drop worker data

bitphage committed:
Add small sleep when dropping state from cli

Database worker is threaded and don't have a chance to exec a noreturn
query, so small time is needed.

bitphage committed:
Add preventive check for node sync

joelvai committed:
Change host_ip back to 8.8.8.8

joelvai committed:
Merge remote-tracking branch 'origin/nodelist-reset' into pr-710-merge

joelvai committed:
Change dexbot version number to 0.18.0

joelvai committed:
Merge remote-tracking branch 'thehapax/cancel-bulk-orders' into pr-701-merge

joelvai committed:
Change dexbot version number to 0.18.1

joelvai committed:
Remove brew update

bitphage committed:
Merge pull request #706 from froooze/patch-2

make lower/upper bound more clear

bitphage committed:
Add note on installing pre-commit hooks

bitphage committed:
Reformat code with black & isort

No pep8 fixes, just reformat.

Closes: #716

bitphage committed:
Make timeout to be optional

Closes: #686

bitphage committed:
Update bitshares libraries

Closes: #692

bitphage committed:
Remove redundant code

bitphage committed:
Define self.last_check in StrategyBase

Closes: #566

bitphage committed:
Switch to use check_last_run in RO

bitphage committed:
Use check_last_run in SO

bitphage committed:
Use check_last_run in KOTH

bitphage committed:
Switch node on tapos_block_summary exception

For details oh why and how it's happening, see
https://github.com/xeroc/python-graphenelib/issues/144

Should not happen in new release of python-graphenelib.
Node switch is still needed in case of unexpected reconnection to node
out of sync.

Closes: #680

bitphage committed:
Fix error() in strategy template

Closes: #473

bitphage committed:
Adjust inaccuracy for full-sized closer order

New algo allows 5% difference for closer order instead of fixed amount

Closes: #739

bitphage committed:
Cleanup txbuffer if worker got disabled

Closes: #505

bitphage committed:
Wrap trx execution via retry_action()

This is needed to add handling of common broadcast errors.

Closes: #748

joelvai committed:
Change dexbot version number to 0.19.0

joelvai committed:
Merge remote-tracking branch 'vvk/686-gui-fail-sorted-nodelist' into pr-718-merge

joelvai committed:
Change dexbot version number to 0.19.1

joelvai committed:
Merge pull request #719 from bitfag/tip-fix-docs

Small docs fixes

joelvai committed:
Merge remote-tracking branch 'vvk/fix-so-price-inversion' into pr-720-review

joelvai committed:
Change dexbot version number to 0.19.2

joelvai committed:
Merge remote-tracking branch 'vvk/fix-test_maintain_strategy_one_sided' into pr-721-merge

joelvai committed:
Change dexbot version number to 0.19.3

joelvai committed:
Merge remote-tracking branch 'vvk/692-libs-update' into pr-722-merge

joelvai committed:
Change dexbot version number to 0.20.0

joelvai committed:
Merge remote-tracking branch 'vvk/tip-disable-fallback-so' into pr-723-merge

joelvai committed:
Change dexbot version number to 0.21.0

joelvai committed:
Merge remote-tracking branch 'vvk/tip-so-tests-workaround' into pr-725-merge

joelvai committed:
Change dexbot version number to 0.21.1

joelvai committed:
Merge remote-tracking branch 'vvk/554-so-asset-threshold' into pr-726-merge

joelvai committed:
Change dexbot version number to 0.21.2

joelvai committed:
Merge remote-tracking branch 'vvk/fix-order-reconstruction' into pr-727-merge

joelvai committed:
Change dexbot version number to 0.20.3

joelvai committed:
Merge remote-tracking branch 'vvk/tip-check-node-sync' into pr-728-merge

joelvai committed:
Change dexbot version number to 0.20.4

joelvai committed:
Merge remote-tracking branch 'vvk/tip-cli-drop-worker-data' into pr-729-merge

joelvai committed:
Change dexbot version number to 0.21.6

joelvai committed:
Merge remote-tracking branch 'vvk/tip-so-stop-loss' into pr-730-merge

joelvai committed:
Change dexbot version number to 0.22.0

joelvai committed:
Merge remote-tracking branch 'vvk/680-ref_block_prefix' into pr-732-merge

joelvai committed:
Change dexbot version number to 0.22.1

joelvai committed:
Merge remote-tracking branch 'vvk/473-strategy-template' into pr-744-merge

joelvai committed:
Change dexbot version number to 0.22.2

joelvai committed:
Merge remote-tracking branch 'vvk/739-fix-so-test' into pr-745-merge

joelvai committed:
Change dexbot version number to 0.22.3

joelvai committed:
Merge remote-tracking branch 'vvk/505-txbuffer-reset' into pr-746-merge

joelvai committed:
Change dexbot version number to 0.22.4

bitphage committed:
Merge remote-tracking branch 'upstream/devel' into tip-check_last_run-decorator

joelvai committed:
Merge remote-tracking branch 'vvk/748-so-broadcast-retry-action' into pr-750-merge

joelvai committed:
Change dexbot version number to 0.22.5

joelvai committed:
Merge remote-tracking branch 'vvk/tip-check_last_run-decorator' into pr-731-merge

joelvai committed:
Change dexbot version number to 0.23.0

bitphage committed:
Require config instance in ConfigValidator

Now ConfigValidator doesn't have to instantiate config on its own, so it
doesn't looses custom config.

Closes: #631

bitphage committed:
Fix ConfigValidator init in WorkerController

bitphage committed:
Adjust 0 balance message in "both" mode

To not scary user when some asset has 0 balance in "both" mode, change
log message level to debug.

Despite report, such condition doesn't leads to "worker disabled" error.

Closes: #715

bitphage committed:
Add bitasset fixtures from python-bitshares

bitphage committed:
Move default account name to fixture

bitphage committed:
Add support for margin call orders in KOTH

Closes: #677

joelvai committed:
Merge remote-tracking branch 'vvk/todo/677-koth-margin-calls' into pr-754-merge

joelvai committed:
Change dexbot version number to 0.24.0

joelvai committed:
Merge branch 'devel' into pr-754-merge

joelvai committed:
Change dexbot version number to 0.24.1

joelvai committed:
Merge remote-tracking branch 'vvk/done/715-koth-trade-0-balance' into pr-755-merge

joelvai committed:
Change dexbot version number to 0.24.2

bitphage committed:
Add initial test for BitsharesOrderEngine

bitphage committed:
Refactor __init__ of StrategyBase and BitsharesOrderEngine

bitphage committed:
Mark primitives __init__ tests as mandatory

bitphage committed:
Add test for invert= in place_market_sell_order()

bitphage committed:
Add more pre-commit hooks

bitphage committed:
Apply autoflake

bitphage committed:
Run docformatter on all files

bitphage committed:
Run isort on all files

bitphage committed:
Fix some mypy errors

bitphage committed:
Remove balances() from StrategyBase

It's defined in BitsharesOrderEngine

bitphage committed:
Fix methods order

bitphage committed:
Move call outside for cycle

bitphage committed:
Add generic method to get operation balance

Closes: #747

bitphage committed:
Fix flake8 errors

bitphage committed:
Respect operational percent settings in RO

Closes: #712

bitphage committed:
Fix flake8 errors

bitphage committed:
Respect operational percent settings in KOTH

bitphage committed:
Mark all storage tests as mandatory

bitphage committed:
Refactor DatabaseWorker singleton

Instead of having single DatabaseWorker instance across all application,
keep one instance per each sqlite database file. Now we can init Storage
with custom sqlite database if needed.

Closes: #667

bitphage committed:
Add test to repruduce #601

bitphage committed:
Update limiting logic in SO

New logic correctly handles #601 corner case, which happens when spread
is not reached, sides imbalanced, several sell orders filled at once and
need to place closer buy order. To correctly determine limiter, just use
previously stored orders (fetch from db).

Updated logic is more correctly determines limits for mountain and
neutral modes also.

Closes: #601

bitphage committed:
Remove spread check from test_maintain_strategy_one_sided

Not applicable with get_actual_spread()

bitphage committed:
Fix SO tests failed due to min_check_interval

Regression introduced in ed2ef04cb4969d0db0e7edda7bd4e457184ccd73

bitphage committed:
Add debug on asset issue

bitphage committed:
Add workarounds for some SO tests

Sevaral tests were market as xfailed due to imprecise price caclulation,
this commit adds pytest.approx() to allow slight error, see #575

bitphage committed:
Disallow 0-profit spread/increment

Profit from one filled order in SO should be > 0, previously >= 0 was
allowed.

bitphage committed:
Use only one market to run SO tests

Looks like two different markets with flipped asset precisions is
redundant and doubles tests run time. Leave only one market for speed
up.

bitphage committed:
Use separate assets for account_1_sat

When all tests run at once, amount to issue may excess max supply

bitphage committed:
Remove unused logger

bitphage committed:
Use separate worker to simulate foreign trader

Fixture-based approach (other_worker) is needed to correctly teardown
tests not leaving open orders.

git bisect note: all SO tests are passing

bitphage committed:
Set formatter for dexbot.per_worker in SO tests

bitphage committed:
Fix clean exit from cli

See https://github.com/websocket-client/websocket-client/issues/449

Closes: #594

joelvai committed:
Change dexbot version number to 0.25.0

joelvai committed:
Merge remote-tracking branch 'vvk/done/pre-commit-additions' into pr-760-merge

joelvai committed:
Change dexbot version number to 0.26.0

joelvai committed:
Merge remote-tracking branch 'vvk/wip/667-storage-db-path' into pr-761-merge

joelvai committed:
Change dexbot version number to 0.27.0

joelvai committed:
Merge remote-tracking branch 'vvk/done/so-imbalanced-several-orders-filled' into pr-762-merge

joelvai committed:
Change dexbot version number to 0.27.1

joelvai committed:
Merge remote-tracking branch 'vvk/done/594-cli-exit' into pr-763-merge

joelvai committed:
Change dexbot version number to 0.28.0

bitphage committed:
Set semantic-versioned deps in requirements-dev.txt

bitphage committed:
Remove all indirect requirements

bitphage committed:
Move pyinstaller dep to requirements-dev.txt

bitphage committed:
Set main requirements to semantically-versioned

Closes: #767

bitphage committed:
Read requirements.txt inside setup.py

bitphage committed:
Bump bitshares dependency

Also remove direct graphenelib dependency, we don't need it.

bitphage committed:
Restore websocket-client dependency

bitphage committed:
Install devel requirements before running pyinstaller

bitphage committed:
Remove websocket-client dependency

Required version is conflicting with python-graphenelib dependency.

Reopen #594

bitphage committed:
Stick to pyqt<5.14

Travis cannot build 5.14.x due to old pip version, the error:

AttributeError: module 'sipbuild.api' has no attribute
'prepare_metadata_for_build_wheel'

See https://www.riverbankcomputing.com/pipermail/pyqt/2020-January/042430.html

bitphage committed:
Bump pytest dependency

bitphage committed:
Fix compatibility with pyinstaller 3.6

Old hook name Crypto was in conflict with built-in pyinstaller hook, so
rename local hook.

bitphage committed:
Fix travis build

bitphage committed:
Fix insufficient balance error

Case happens when base or quote balance is not enough to place minimum
allowed order when placing lowest buy or highest sell order.

Closes: #765

bitphage committed:
Handle null trx['operation_results'] in SO

Closes: #764

bitphage committed:
Apply docformatter fix

joelvai committed:
Remove deprecated configurations from .travis.yml

sudo is deprecated and simply has no effect anymore.

skip_cleanup is deprecated and replaced with cleanup, which is false by
default.

joelvai committed:
Change dexbot version number to 0.28.1

joelvai committed:
Merge remote-tracking branch 'vvk/done/767-semantic-deps' into pr-768-merge

joelvai committed:
Change dexbot version number to 0.29.0

joelvai committed:
Merge remote-tracking branch 'vvk/done/765-so-insufficient-balance' into pr-770-merge

joelvai committed:
Change dexbot version number to 0.29.1

joelvai committed:
Merge remote-tracking branch 'vvk/done/764-so-none-type' into pr-771-merge

joelvai committed:
Change dexbot version number to 0.29.2

joelvai committed:
Change dexbot version number to 1.0.0

joelvai committed:
Change .travis.yml

Reverted back some settings

bitphage committed:
Update README.md

Add reference to active fork.
